Creating new azure templates for containerapps

twerthi opened the pull request:
---

# Background

Creating new templates to support deployments to Azure Container Apps.

1. Template to create a Container App Environment if it doesn't exist and set an output variable for it's ID
2. Template to deploy a Container App

# Results

Automates the deployment of an Azure Container app.

## Before

<!-- Consider adding a log excerpt or screen capture. -->

## After

<!-- Consider adding a log excerpt or screen capture. -->

# Pre-requisites

- [X] `Id` should be a **GUID** that is not `00000000-0000-0000-0000-000000000000`
  - **NOTE** If you are modifying an existing step template, please make sure that you **do not** modify the `Id` property *(updating the `Id` will break the Library sync functionality in Octopus)*. 
- [X] `Version` should be incremented, otherwise the integration with Octopus won't update the step template correctly
- [X] Parameter names should not start with `$`
- [X] **Step template parameter names (the ones declared in the JSON, not the script body) should be prefixed with a namespace so that they are less likely to clash with other user-defined variables in Octopus** (see [this issue](https://github.com/OctopusDeploy/Issues/issues/2126)). For example, use an abbreviated name of the step template or the category of the step template).
- [X] `LastModifiedBy` field must be present, and (_optionally_) updated with the correct author
- [ ] If a new `Category` has been created:
   - [ ] An image with the name `{categoryname}.png` must be present under the `step-templates/logos` folder
   - [ ] The `switch` in the `humanize` function in [`gulpfile.babel.js`](https://github.com/OctopusDeploy/Library/blob/master/gulpfile.babel.js#L92) must have a `case` statement corresponding to it

benPearce1 reviewed (approved):
LGTM, didn't test.

One suggestion:
```
# Check to see if $IsWindows is available
if ($null -eq $IsWindows)
{
     Write-Host "Determining Operating System..."
    switch ([System.Environment]::OSVersion.Platform)
    {
         "Win32NT"
        {
             # Set variable
            $IsWindows = $true
            $IsLinux = $false
        }
        "Unix"
        {
             $IsWindows = $false
            $IsLinux = $true
        }
    }
}
```

can be replaced with:
```
# Check to see if $IsWindows is available
if ($null -eq $IsWindows)
{
     Write-Host "Determining Operating System..."
    $IsWindows = [System.Environment]::OSVersion.Platform == "Win32NT"
    $IsLinux = [System.Environment]::OSVersion.Platform == "Unix"
}
```

benPearce1 commented:
Failed build: `Failed: Package FeedId for azure-deploy-containerapp.json should be null, but was: Feeds-2717`

twerthi commented:
The double `==` doesn't work in PowerShell (or at least it didn't on my machine).  Updated the code, implemented suggested changes (updated to work on my machine of course) as well as fixed a bug in adding the saved powershell module where it wasn't using the OS specific path separator to make it cross-platform compatible.
